mlrun/datastore/base.py

```diff
@@ -36,6 +39,141 @@
 from .utils import filter_df_start_end_time, select_columns_from_df
 
 
+def _get_pyarrow_version():
+    """Get the current PyArrow version."""
+    return version.parse(pyarrow.__version__)
+
+
+def is_pyarrow_17_or_later():
+    """Check if PyArrow version is 17.0.0 or later."""
+    return _get_pyarrow_version() >= version.parse("17.0.0")
+
+
+def _detect_partition_data_conflict(error_message):
+    """
+    Detect if the error is a PyArrow 17+ schema conflict between partition and data columns.
+
+    Args:
+        error_message (str): The error message from PyArrow
+
+    Returns:
+        dict: Contains conflict information or None if not a conflict
+    """
+    # Only check for conflicts in PyArrow 17+
+    if not is_pyarrow_17_or_later():
+        return None
+
+    # Pattern to match PyArrow schema conflict errors
+    conflict_pattern = (
+        r"Unable to merge: Field (\w+) has incompatible types: (.+?) vs (.+)$"
```

Comment:
This is probably too fragile. Any change to the error message in PyArrow will break it.